src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -106,20 +108,20 @@ def generate_grafana_ds_config(hosts: List[str]) -> str:
         return config_file, sorted(deps)
 
     def daemon_check_post(self, daemon_descrs: List[DaemonDescription]):
-        # make sure the dashboard [does not] references grafana
-        try:
-            current_url = self.mgr.get_module_option_ex('dashboard', 'GRAFANA_API_URL')
-            host = daemon_descrs[0].hostname
-            url = f'https://{self.mgr.inventory.get_addr(host)}:3000'
-            if current_url != url:
-                logger.info('Setting dashboard grafana config to %s' % url)
-                self.mgr.set_module_option_ex('dashboard', 'GRAFANA_API_URL', url)
-                # FIXME: is it a signed cert??
-        except Exception as e:
-            logger.debug('got exception fetching dashboard grafana state: %s', e)
-
+        # TODO: if there are multiple daemons, who is the active one?
+        # TODO: signed cert
+        dd = daemon_descrs[0]
```

Comment:
ok I need this very soon. Can you please encapsulate this? 

```python

def get_active_daemon(self, dds: List[DaemonDescription]) -> DaemonDescription: ...
```

Reply:
Done.